qutip/tests/test_gates.py

```diff
@@ -53,7 +53,7 @@ def _remove_global_phase(qobj):
     Return a new Qobj with the gauge fixed for the global phase.  Explicitly,
     we set the first non-zero element to be purely real-positive.
     """
-    flat = qobj.full().flat.copy()
+    flat = qobj.tidyup(1e-14).full().flat.copy()
```

Comment:
Wondering why this got set explicitly here? Not a bad idea to set it explicitly, just wondering if the test failed without it.

Reply:
The matrices here can be `Dense`, and it set the phase to the first non-zero.
So without the `auto-tidyip` in `Qobj`, the first element could be 1e-30 and the global phase would not be properly set.